core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/EraSubmissionPreparation.java

```diff
@@ -0,0 +1,496 @@
+
+      Element s = submissionDocument.createElementNS(null, "SUBMISSION");
+      s.setAttribute("alias", submission.getAlias());
+      s.setAttribute("submission_date", DF_TIMESTAMP.format(submission.getSubmissionDate()));
+      s.setAttribute("submission_comment", submission.getDescription());
+      s.setAttribute("center_name", centreName);
+
+      Element title = submissionDocument.createElementNS(null, "TITLE");
+      title.setTextContent(submission.getTitle());
+      s.appendChild(title);
+
+      Element contacts = submissionDocument.createElementNS(null, "CONTACTS");
+      Stream
+          .concat(Stream.of(user),
+              submission.getExperiments().stream().map(experiment -> experiment.getSecurityProfile().getOwner()).filter(Objects::nonNull))
+          .map(User::getFullName).distinct()
+          .map(contactName -> {
+            Element contact = submissionDocument.createElementNS(null, "CONTACT");
+            contact.setAttribute("name", contactName);
+            return contact;
+          }).forEach(contacts::appendChild);
+
+      s.appendChild(contacts);
+
+      Element actions = submissionDocument.createElementNS(null, "ACTIONS");
+
+      for (ChildSubmissionFile subFile : FILES) {
+
+        subFile.generateDocument(output);
+        if (!subFile.isEmpty()) {
+          if (submissionAction == SubmissionActionType.ADD || submissionAction == SubmissionActionType.VALIDATE) {
+            Element action = submissionDocument.createElementNS(null, "ACTION");
+            Element validate = submissionDocument.createElementNS(null, submissionAction.name());
+            validate.setAttribute("schema", subFile.name());
+            validate.setAttribute("source", subFile.fileName());
+            action.appendChild(validate);
+            actions.appendChild(action);
+          }
+        }
+      }
+      s.appendChild(actions);
+
+      if (submissionDocument.getElementsByTagName("SUBMISSION_SET").item(0) != null) {
+        submissionDocument.getElementsByTagName("SUBMISSION_SET").item(0).appendChild(s);
+      } else {
+        Element submissionSet = submissionDocument.createElementNS(null, "SUBMISSION_SET");
+        submissionDocument.appendChild(submissionSet);
+        submissionSet.appendChild(s);
+      }
+
+      ZipEntry mainFile = new ZipEntry("SUBMISSON.xml");
+      output.putNextEntry(mainFile);
+      Transformer transformer = TransformerFactory.newInstance().newTransformer();
+      transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+
+      StreamResult result = new StreamResult(output);
+      DOMSource source = new DOMSource(submissionDocument);
+      transformer.transform(source, result);
+      output.closeEntry();
+      output.close();
+      return outputBytes.toByteArray();
+
+    } catch (ParserConfigurationException | TransformerException | IOException e) {
+      throw new RuntimeException(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)